### PR TITLE
Resolved the basics of issue #37 - improving the nav bar

### LINKED ISSUE #37: Improvise the navbar with drop downs for each navbar item
Develop a fully responsive navigation bar with hoverable dropdown menus for the **"Donations", "Donors",** and **"Programs"** sections of our website. Each dropdown menu should contain specific actions and ensure smooth navigation to corresponding pages upon selection.

Image

**Requirements:**

- [ ] **Navbar Structure:**

- Update the navigation bar to include three main items: Donations, Donors, and Programs.
- When hovering over any of these items, display a dropdown menu directly below the hovered item.

- [ ] **Dropdown Menus:**

- Each main item in the navbar should have a dropdown menu that appears on hover.
- The dropdown for "Donations" should include links to "All Donations" and "Add New Donation".
- The dropdown for "Donors" should include links to "All Donors" and "Add New Donor".
- The dropdown for "Programs" should include links to "All Programs" and "Add New Program".

- [ ] **Styling:**

- Use CSS for styling the navbar and dropdowns with a professional and clean aesthetic that matches the website’s existing design.
- Implement hover effects for dropdown items to enhance user experience.
- Ensure the navbar is responsive and adapts to different screen sizes.

### PR DESCRIPTION
**Pull request by truffer11**
Fixes #37 

### What was changed?

- The navigation bar has been switched from javascript to typescript.
- The navigation bar at the top was modified to allow for dropdown options.
- Donations, programs, and donors have all received dropdown options.
- All buttons that lead to the same place as a dropdown option have been removed.

### Why was it changed?

- The navigation bar was switched from javascript to typescript for security and to work towards our new standard of rewriting all javascript into typescript.
- Navigating the web page could be confusing to the user, so dropdown menus should hopefully simplify that.
- The new dropdowns also allow for quicker access to certain pages, in contrast from having to load several pages, find the right button, and load another page.

### How was it changed?

- Added a new .css file, NavbarDropdown.css, which deals with the styling of the dropdown menus
- Rewrote the entire Navbar.js into Navbar.tsx. (NavbarOld.js will be deleted soon, kept for screenshots)
- Navbar.tsx deals with the logic of the navigation bar
- Programs.tsx and DonatedItemsList.js have had their respective add buttons and functions removed

### Screenshots that show the changes (if applicable):

- **Before:**
 
Screenshot 2024-10-14 153300

- **After:**

Screenshot 2024-10-14 153440
Screenshot 2024-10-14 154033
Screenshot 2024-10-14 154051

### CODE REVIEW

File: client-app/src/Components/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+
+const Navbar: React.FC = () => {
+    const [user, setUser] = useState<string>('');
+    const location = useLocation();
+    const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
+
+    useEffect(() => {
+        const token = localStorage.getItem('token');
+        const name = localStorage.getItem('name');
```
Comment (miabu-pashh) [inline client-app/src/Components/Navbar.tsx:13]: Remove the isLogged item from localStorage and rely solely on token to determine the logged-in state.

File: client-app/src/Components/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    const handleLogout = (): void => {
+        localStorage.removeItem('token');
+        localStorage.removeItem('name');
+        setUser('');
+        localStorage.setItem('isLogged', 'false');
+        setIsLoggedIn(false);
+
+        window.location.href = '/';
+    };
```
Comment (miabu-pashh) [inline client-app/src/Components/Navbar.tsx:35]: In handleLogout, you can simplify by removing only the token and name

File: client-app/src/Components/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                                            color: location.pathname === '/about' ? 'black' : 'inherit',
+                                        }}
+                                    >
+                                        About
+                                    </Link>
+                                </li>
+                            </>
+                        ) : (
+                            <>
+                                <li className="nav-item" style={{ fontSize: '20px', paddingLeft: '10px' }}>
```
Comment (miabu-pashh) [inline client-app/src/Components/Navbar.tsx:144]: better with separate the inline styles with css file.

File: client-app/src/Components/Programs.tsx [hunk 4]
```diff
@@ -35,11 +35,6 @@ function Programs() {
     return (
         <div className="programs-container">
             <h1>Programs</h1>
-            <div className="add-program">
```
Comment (miabu-pashh) [inline client-app/src/Components/Programs.tsx:38]: removed the add program,  added in the nav bar dropdown

**Review by miabu-pashh — COMMENTED** [review]
- Removing the logged-Item , and rely on token in logged in state
- handlelogout simplification,
- usage of css class more than inline styles
- better added the add program in dropdown menu

**Review by Anjali0407-git — CHANGES_REQUESTED** [review]

File: client-app/src/Components/NavbarOld.js
Comment (Anjali0407-git) [inline client-app/src/Components/NavbarOld.js:1]: remove this file

File: client-app/src/Components/Programs.tsx [hunk 5]
```diff
@@ -35,11 +35,6 @@ function Programs() {
...
-                <Link to="/addprogram">
-                    <button className="add-program-button">Add Program</button>
-                </Link>
-            </div>
             <div className="program-list">
```
Comment (Anjali0407-git) [inline client-app/src/Components/Programs.tsx:38]: Similar to `Add new Donation` button above, undelete the `Add Program` button on programs page for easy access

**Review by Anjali0407-git — CHANGES_REQUESTED** [review]

File: client-app/src/Components/DonatedItemsList.js [hunk 6]
```diff
@@ -404,9 +390,6 @@ function DonatedItemsList() {
                 <div
                     style={{ position: 'fixed', bottom: '20px', right: '20px' }}
                 >
-                    <button onClick={() => handleAddDonationClick()}>
-                        <FaPlus size={24} />
-                    </button>
                 </div>
```
[the remark this answers is not in the document]
Comment (truffer11) [reply client-app/src/Components/DonatedItemsList.js]: I think I like the idea of having two buttons for that reason. I've also updated the plus button to go to our new "Add New Donation" form. It was linking to some form I have never seen, but has effectively been replaced by the "Add New Donation" form from the last sprint. I will probably go through and find where else this old form is.

File: client-app/src/Components/Navbar.tsx (hunk 1 above)
Comment (truffer11) [reply client-app/src/Components/Navbar.tsx:13]: I briefly tried to implement this, however it did not allow me to do signed in functionality when I relied only on token. I will try again at a later time

**Pull request merged**